addons/auth_signup/models/res_partner.py

```diff
@@ -105,6 +105,9 @@ def signup_get_auth_param(self):
         """ Get a signup token related to the partner if signup is enabled.
             If the partner already has a user, get the login parameter.
         """
+        if not self.env.user.has_group('base.group_user') and not self.env.is_admin():
```

Comment:
Isn't it bettet to use the ``assert_log_admin_access`` decorator, to:
- Check the admin access
- Log the access

BTW, the condition seems odd.

Reply:
@tivisse But this decorator will not check if the current user is an internal user (`has_group('base.group_user')`)

Reply:
The idea is to avoid the portal user from calling this method, so he can't generate user linked to the wanted partner (and then to read/write on it)

Reply:
ok so we authorize the sudoed portal users to access this method. 

Thanks for the explanation.